### PR TITLE
feat: Add git shallow clone to FAQ

### PR DESCRIPTION
**Pull request by Filbird**
## Context
Now that git shallow cloning is performed by default, it should be more clearly documented in the guide.

## Objectives
Add git shallow clone to FAQ

## References
https://github.com/screwdriver-cd/screwdriver/projects/4#card-14436792

### CODE REVIEW

**Review by pranavrc — APPROVED** [review]
image

**Review by minzcmu — APPROVED** [review]

File: docs/user-guide/FAQ.md [hunk 1]
```diff
@@ -128,3 +128,10 @@ By default, step commands are evaluated with the Bourne shell (`/bin/sh`). You c
 
 ## How do I speed up time to upload artifacts?
 You can set the environment variable [`SD_ZIP_ARTIFACTS`](./environment-variables#user-configurable) to `true` which will zip artifacts before uploading, provided your cluster admin has set it up properly.
+
+## How do I disable shallow cloning?
+You can set the environment variable [`GIT_SHALLOW_CLONE`](./environment-variables#user-configurable) to `false` in order to disable shallow cloning.
+
+By default Screwdriver will shallow clone source repositories to a depth of 50. Screwdriver will also enable the `--no-single-branch` flag by default.
```
Comment (jithine) [inline docs/user-guide/FAQ.md:137]: should this be behind a config ?

Comment (Filbird) [reply docs/user-guide/FAQ.md:137]: I'm not seeing any noticeable difference whether this flag is enabled or disabled when testing a large repo (~160k commits).

Comment (Filbird) [reply docs/user-guide/FAQ.md:137]: On a separate note, the branching logic for `getCheckoutCommand` is slowly getting more complex. At some point we may want to consider refactoring the way we approach bookends.

**Review by tkyi — APPROVED** [review]

**Pull request merged**